lib/countly.js

```diff
@@ -1710,67 +1710,78 @@
             });
         }
     };
+
+    /**
+     * Prepare widgets according current options
+     * @param {array=} - optional - widget ids array
+     */
+    Countly.initializeFeedbackPopups = function(enableWidgets) {
+        if (!enableWidgets) {
+            enableWidgets = store('cly_fb_widgets');
+        }
+        sendXmlHttpRequest(Countly.url + '/o/feedback/multiple-widgets-by-id', {widgets:JSON.stringify(enableWidgets)}, function(err, params, responseText){
+            if(err){
+                log("Errors occurred:", err);
```

Comment:
Missing return?

Reply:
Also even on error, we still need to get to processWidgets function to remove old widgets that are not valid for this page, right?

Reply:
thanks for comments. added missing return and moved function remove section into initializeFeedbackPopups method from processWidget.